### PR TITLE
Fix(eos_cli_config_gen): Various fixes for router path-selection & application-traffic-recognition

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Some fixes in Jinja2 templates for empty lists that were overlooked

## Related Issue(s)

Discovered while testing trailblazer

## Component(s) name

`arista.avd.eos_cli_config_gen`

## How to test

molecule tests

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2 [hunk 1]
```diff
@@ -23,8 +23,8 @@
 {%                 set tcp_dest_port = application.tcp_dest_port_set_name | arista.avd.default("-") %}
 {%                 set udp_src_port = application.udp_src_port_set_name | arista.avd.default("-") %}
 {%                 set udp_dest_port = application.udp_dest_port_set_name | arista.avd.default("-") %}
-{%                 set protocol_ranges = ", ".join(application.protocol_ranges) | arista.avd.default("-") %}
-{%                 set protocol = ", ".join(application.protocols) | arista.avd.default("-") %}
+{%                 set protocol_ranges = ", ".join(application.protocol_ranges | arista.avd.default([])) | arista.avd.default("-") %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2]:
```suggestion
{%                 set protocol_ranges = ", ".join(application.protocol_ranges | arista.avd.default(['-'])) %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2 [hunk 2]
```diff
@@ -23,8 +23,8 @@
...
+{%                 set protocol = ", ".join(application.protocols | arista.avd.default([])) | arista.avd.default("-") %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2]:
```suggestion
{%                 set protocol = ", ".join(application.protocols | arista.avd.default(['-'])) %}
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/application-traffic-recognition.md [hunk 3]
```diff
@@ -99,6 +103,8 @@ interface Management1
 | ---- | -------- |
 | dest_prefix_set1 | 2.3.4.0/24 |
 | dest_prefix_set2 | 4.4.4.0/24 |
+| empty-ipv4-prefixes |  |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/application-traffic-recognition.md]: My suggestions should fix these blank fields.

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/application-traffic-recognition.md]: hmm this is some other place - will need to do the same

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/application-traffic-recognition.md]: [7570f90](https://github.com/aristanetworks/ansible-avd/pull/3504/commits/7570f906d9b50a4a1676a765ea88fd3a971b4966)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2 (hunk 2 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/application-traffic-recognition.j2]: [7570f90](https://github.com/aristanetworks/ansible-avd/pull/3504/commits/7570f906d9b50a4a1676a765ea88fd3a971b4966)

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested in lab

**Pull request merged**